pkg/manager/member/tidb_member_manager.go

```diff
@@ -241,10 +249,103 @@ func (m *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.TidbC
 			return err
 		}
 	}
+	// set random password
+	if tc.Spec.TiDB.Initializer != nil && tc.Spec.TiDB.Initializer.CreatePassword && !tc.Status.TiDB.InitPasswordPhase && !tc.IsTLSClusterEnabled() {
+		// check tidb pod is ready
+		podOrdinals := helper.GetPodOrdinals(*oldTiDBSet.Spec.Replicas, oldTiDBSet).List()
+		isTiDBReady := true
+		for _i := len(podOrdinals) - 1; _i >= 0; _i-- {
+			i := podOrdinals[_i]
+			podName := tidbPodName(tcName, i)
+			pod, err := m.deps.PodLister.Pods(ns).Get(podName)
+			if err != nil {
+				return fmt.Errorf("failed to get pods %s for cluster %s/%s, error: %s", podName, ns, tcName, err)
+			}
+			isReady := podutil.IsPodReady(pod)
+			if !isReady {
+				isTiDBReady = false
+				break
+			}
+		}
+		if isTiDBReady {
+			// sync password secret
+			var password string
+			secretName := controller.TiDBSecret(tc.Name)
+			secret, err := m.deps.SecretLister.Secrets(ns).Get(secretName)
+			passwordSecretExist := true
+			if err != nil {
+				if errors.IsNotFound(err) {
+					passwordSecretExist = false
+				} else {
+					return err
+				}
+			}
+
+			if !passwordSecretExist {
+				klog.Infof("Create random password for cluster[%s:%s]", tc.Namespace, tc.Name)
+				var secret *corev1.Secret
+				secret, password = m.buildRandomPasswordSecret(tc)
+				err := m.deps.TypedControl.Create(tc, secret)
+				if err != nil {
+					return err
+				}
+			} else {
+				password = string(secret.Data[constants.TidbRootKey])
+			}
+			// init password
+			var db *sql.DB
+			var dsn string
+			err = wait.PollImmediate(1*time.Second, 5*time.Second, func() (done bool, err error) {
+				dsn, err = m.GetDBUrl(tc)
+				if err != nil {
+					klog.Errorf("can't get dsn of tidb cluster[%s:%s], err: %s", tc.Namespace, tc.Name, err)
+					return false, err
+				}
+				ctx := context.TODO()
+				db, err = m.OpenDB(ctx, dsn)
+				if err != nil {
+					klog.Warningf("can't connect to tidb cluster[%s:%s], err: %s", tc.Namespace, tc.Name, err)
+					if ctx.Err() != nil {
+						return false, ctx.Err()
+					}
+					return false, nil
+				}
+				return true, nil
+			})
+			if err != nil {
+				klog.Errorf("can't get connection of tidb cluster[%s:%s], err: %s", tc.Namespace, tc.Name, err)
+			} else {
+				err = m.SetPassword(context.TODO(), db, password)
+				if err != nil {
+					klog.Errorf("set tidb[%s:%s] password err: %s", tc.Namespace, tc.Name, err)
+				}
+				tc.Status.TiDB.InitPasswordPhase = true
+				klog.Infof("set password successfully for tidb[%s:%s]", tc.Namespace, tc.Name)
+			}
+			defer db.Close()
+		} else {
+			klog.Infof("set password wait for tidb[%s:%s] pod ready", tc.Namespace, tc.Name)
```

Comment:
Is this necessary?

Reply:
I think we need to print a log when set successfully